lib/vintage_net/connectivity/ssl_connect.ex

```diff
@@ -0,0 +1,64 @@
+defmodule VintageNet.Connectivity.SSLConnect do
+  @moduledoc """
+  Test connectivity by making a connection using SSL
+
+  Connectivity with a remote host can be checked by making a SSL connection to
+  it. The connection either works, the connection is refused, or it times out.
+  The first two cases indicate connectivity.
+  """
+
+  import VintageNet.Connectivity.TCPPing, only: [get_interface_address: 2]
+  alias VintageNet.Connectivity.HostList
+  require Logger
+
+  @connect_timeout 5_000
+
+  @doc """
+  Check connectivity with another device
+
+  The "connect" is a SSL connection attempt from the specified interface to
+  an IP address and port. Failures to connect don't necessarily mean that the
+  Internet is down, but it's likely especially if the server that's specified
+  in the configuration is highly available.
+  """
+  @spec connect(VintageNet.ifname(), HostList.options()) :: :ok | {:error, :inet.posix()}
+  def connect(ifname, opts) do
+    host = Keyword.fetch!(opts, :host)
+    port = Keyword.fetch!(opts, :port)
+
+    with {:ok, src_ip} <- get_interface_address(ifname, :inet),
+         {:ok, ssl} <-
+           :ssl.connect(
+             to_charlist(host),
+             port,
+             connect_opts(src_ip),
+             @connect_timeout
+           ) do
+      _ = :ssl.close(ssl)
+      :ok
+    else
+      {:error, reason} ->
+        {:error, reason}
+
+      posix_error ->
+        {:error, posix_error}
+    end
+  end
+
+  defp connect_opts(src_ip) do
+    base = [
+      verify: :verify_peer,
+      active: false,
+      ip: src_ip
+    ]
+
+    if Code.ensure_loaded?(:public_key) and function_exported?(:public_key, :cacerts_get, 0) do
+      cacerts = apply(:public_key, :cacerts_get, [])
+      Keyword.put(base, :cacerts, cacerts)
+    else
+      Logger.warning("SSLConnect support on OTP 24 is limited due to lack of cacerts")
+      # remove the verify_peer option, since we don't have CA certs
+      Keyword.delete(base, :verify)
+    end
+  end
```

Comment:
```suggestion
  if :erlang.system_info(:otp_release) in [~c"21", ~c"22", ~c"23", ~c"24"] do
    defp connect_opts(src_ip) do
      Logger.warning("SSLConnect support on OTP 24 is limited due to lack of cacerts")      
      [
        active: false,
        ip: src_ip
      ]
    end
  else
    defp connect_opts(src_ip) do
      [
        cacerts: :public_key.cacerts_get()
        verify: :verify_peer,
        active: false,
        ip: src_ip
      ]
    end
  end
```
How about this?

If you like it, then my next comment is to pass the options to `connect_opts/1` and let the user specify `:verify` and :cacerts` so that it's possible to hardcode the acceptable certs. In practice, we'll need to do this because `:public_key.cacerts_get/0` is limited to the certs we have around when building firmware. Those may expire while a device is sitting in a warehouse and prevent internet connectivity from being detected.